tests/manage/z_cluster/test_rook_ceph_log_rotate.py

```diff
@@ -0,0 +1,194 @@
+import logging
+import time
+import pytest
+import re
+
+from ocs_ci.ocs import defaults
+from ocs_ci.ocs.resources.storage_cluster import verify_storage_cluster
+from ocs_ci.utility.utils import TimeoutSampler
+from ocs_ci.ocs.cluster import ceph_health_check
+from ocs_ci.ocs.resources import pod
+from ocs_ci.ocs.exceptions import TimeoutExpiredError
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.ocs import constants
+from ocs_ci.framework.testlib import (
+    ManageTest,
+    tier2,
+    bugzilla,
+    skipif_ocs_version,
+    skipif_external_mode,
+    ignore_leftovers,
+    config,
+)
+
+log = logging.getLogger(__name__)
+
+
+@tier2
+@ignore_leftovers
+@bugzilla("2116416")
+@skipif_external_mode
+@skipif_ocs_version("<4.10")
+@pytest.mark.polarion_id("OCS-4684")
+class TestRookCephLogRotate(ManageTest):
+    """
+    Test Rook Ceph Log Rotate
+
+    """
+
+    @pytest.fixture(autouse=True)
+    def teardown(self, request):
+        def finalizer():
+            log.info("Delete logCollector from storage cluster yaml file")
+            storagecluster_obj = OCP(
+                resource_name=constants.DEFAULT_CLUSTERNAME,
+                namespace=defaults.ROOK_CLUSTER_NAMESPACE,
+                kind=constants.STORAGECLUSTER,
+            )
+            params = '[{"op": "remove", "path": "/spec/logCollector"}]'
+            storagecluster_obj.patch(params=params, format_type="json")
+            log.info(
+                "It takes time for storagecluster to update after the edit command"
+            )
+            time.sleep(30)
+            log.info("Verify storagecluster on Ready state")
+            verify_storage_cluster()
+            ceph_health_check()
```

Comment:
verify_storage_cluster() has a check if storagecluster reached succeeded phase with a timeout of 600 sec.
I think you can remove the hardcoded sleep